src/NzbDrone.Core/Datastore/Migration/139_add_download_history.cs

```diff
@@ -0,0 +1,58 @@
+using System.Data;
+using FluentMigrator;
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(139)]
+    public class add_download_history : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Create.TableForModel("DownloadHistory")
+                  .WithColumn("EventType").AsInt32().NotNullable()
+                  .WithColumn("SeriesId").AsInt32().NotNullable()
+                  .WithColumn("DownloadId").AsString().NotNullable()
+                  .WithColumn("SourceTitle").AsString().NotNullable()
+                  .WithColumn("Date").AsDateTime().NotNullable();
+
+            Create.Index().OnTable("DownloadHistory").OnColumn("EventType");
+            Create.Index().OnTable("DownloadHistory").OnColumn("SeriesId");
+            Create.Index().OnTable("DownloadHistory").OnColumn("DownloadId");
+
+            Execute.WithConnection(InitialImportedDownloadHistory);
+        }
+
+        private void InitialImportedDownloadHistory(IDbConnection conn, IDbTransaction tran)
+        {
+            using (var cmd = conn.CreateCommand())
+            {
+                cmd.Transaction = tran;
+                cmd.CommandText = "SELECT SeriesId, DownloadId, SourceTitle, Date FROM History WHERE DownloadId IS NOT NULL AND EventType = 3 GROUP BY DownloadId";
```

Comment:
Shouldn't EventType = 1 (Grabbed) be processed too?

Reply:
I was more concerned with the imports, I can add the grabs as well.